packages/es-components/src/components/util/Recording.js

```diff
@@ -0,0 +1,111 @@
+// https://help.fullstory.com/hc/en-us/articles/360020623574-How-do-I-protect-my-users-privacy-in-FullStory
+
+import { Children, cloneElement } from 'react';
+import styled from 'styled-components';
+
+const classNameMap = {
+  excluded: 'fs-exclude',
+  masked: 'fs-mask',
+  unmasked: 'fs-unmask'
+};
+
+export const FULLSTORY_FILTER_TYPE = {
+  excluded: 'excluded',
+  masked: 'masked',
+  unmasked: 'unmasked'
+};
+
+function getDisplayName(WrappedComponent) {
+  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+
+function joinClassNames(...classNames) {
+  return classNames.filter(Boolean).join(' ') || '';
+}
+
+function addClassNameToChildren(children, type, props = {}) {
+  return Children.map(children, child =>
+    cloneElement(child, {
+      ...(props ?? {}),
+      className: joinClassNames(child.props?.className, classNameMap[type])
+    })
+  );
+}
+
+export function withRecording(WrappedComponent, type) {
+  // eslint-disable-next-line react/display-name
+  return ({ children, className, ...props }) => {
+    // eslint-disable-next-line no-param-reassign
+    WrappedComponent.displayName = `${type}WithRecording(${getDisplayName(WrappedComponent)})`;
+    return (
+      <WrappedComponent className={joinClassNames(className, classNameMap[type])} {...props}>
+        {children}
+      </WrappedComponent>
+    );
+  };
+}
```

Comment:
Note that these HOCs do not account for fragments. You cannot set a `className` on a fragment.

```
// Single component
return withRecordingUnmasked(<div>...</div>);

// Fragment component - will not work
return withRecordingUnmasked(<>...</>);

```

The HOC functions in this PR need to be adjusted to include siblings.

https://stackoverflow.com/questions/49012718/check-if-a-child-is-react-fragment

```
export function withRecording(WrappedComponent, type) {
  if (ReactIs.isFragment(WrappedComponent)) {
    return { children, ...rest })  => <>{addClassNameToChildren(children, type, rest)}</>;
  }

  return ({ children, className, ...props }) => {
    WrappedComponent.displayName = `${type}WithRecording(${getDisplayName(WrappedComponent)})`;
    return (
      <WrappedComponent className={joinClassNames(className, classNameMap[type])} {...props}>
        {children}
      </WrappedComponent>
    );
  };
}
```